src/io/uring/Ring.cxx

```diff
@@ -10,8 +10,16 @@ namespace Uring {
 Ring::Ring(unsigned entries, unsigned flags)
 {
 	if (int error = io_uring_queue_init(entries, &ring, flags);
-	    error < 0)
-		throw MakeErrno(-error, "io_uring_queue_init() failed");
+	    error < 0) {
+		if (error == -EINVAL && (flags & ~iouring_setup_mask)) {
+		    flags &= iouring_setup_mask;
+			if (error = io_uring_queue_init(entries, &ring, flags);
+			    error < 0)
+				throw MakeErrno(-error, "io_uring_queue_init() failed (fallback)");
```

Comment:
I'm not sure about the `(fallback)` suffix in the error message. I don't think anybody will understand what it means (and the error message is already difficult enough)